memcached/memcached.go

```diff
@@ -9,6 +13,42 @@ type CacheClient struct {
 	client *memcache.Client
 }
 
+// GetBytes converts interface{} to a byte array
+func GetBytes(key interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	if key == nil {
+		return nil, fmt.Errorf("input cannot be nil")
+	}
+	gob.Register(key)
+	err := enc.Encode(&key)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// BytesToEmptyInterface converts byte array to interface{} object
+func BytesToEmptyInterface(data []byte) (interface{}, error) {
+	buf := bytes.NewBuffer(data)
+	dec := gob.NewDecoder(buf)
```

Comment:
check if this can be singleton

Reply:
This can't be singleton since a byte buffer is passed to it which contains the data to be decoded.